USING_ELEKTRA.md

```diff
@@ -0,0 +1,252 @@
+current background color is:
+```sh
+kdb get '/sw/lcdproc/lcdd/#0/current/curses/background'
+#> cyan
+```
+If you want to avoid typing the long path `/sw/lcdproc/lcdd/#0/current` all the time you
+can set a bookmark (see `man kdb`).
+```sh
+kdb set 'user/sw/elektra/kdb/#0/current/bookmarks/lcd' '/sw/lcdproc/lcdd/#0/current'
+kdb get +lcd/curses/background
+#> cyan
+```
+
+The result comes from the set `default` metadata which is saved for the key. If no value is set
+by the user, the `default` value is taken.
+You can see all metadata by calling `kdb lsmeta`:
+```sh
+kdb lsmeta '/sw/lcdproc/lcdd/#0/current/curses/background'
+#> check/enum/#0
+#> check/enum/#1
+#> check/enum/#2
+#> check/enum/#3
+#> check/enum/#4
+#> check/enum/#5
+#> check/enum/#6
+#> check/enum/#7
+#> check/type
+#> default
+#> description
+#> type
+```
+Here we can als see `check/enum/#0-7` which is used by Elektra to assert
+for correct configuration settings. You can query for the values by calling
+`kdb getmeta`
+```sh
+kdb getmeta '/sw/lcdproc/lcdd/#0/current/curses/background' 'check/enum/#0'
+#> red
+```
+This `.../#<number>` notation is the typical array notation which is used by Elektra.
+If you want to query all metadata and see their values you can use this little script:
+```sh
+kdb lsmeta '/sw/lcdproc/lcdd/#0/current/curses/background' \
+| xargs -I% -n1 sh -c 'printf "% = " && kdb getmeta \
+"/sw/lcdproc/lcdd/#0/current/curses/background" "%"'
+#> check/enum = #7
+#> check/enum/#0 = red
+#> check/enum/#1 = black
+#> check/enum/#2 = green
+#> check/enum/#3 = yellow
+#> check/enum/#4 = blue
+#> check/enum/#5 = magenta
+#> check/enum/#6 = cyan
+#> check/enum/#7 = white
+#> default = cyan
+#> description = background color when "backlight" is off
+#> type = string
+```
+In future versions we might include a command which gives you all metadata 
+including their values with a simple command.
+If you try to set the value of the background to an invalid value, Elektra
+will raise an error and prevent a possible misconfiguration:
+```sh
+kdb set '/sw/lcdproc/lcdd/#0/current/curses/background' purple
+#> Using name user/sw/lcdproc/lcdd/#0/current/curses/background
+#> The command kdb set failed while accessing the key database with the info:
+#> Sorry, the error (#121) occurred ;(
+#> Description: validation failed
+#> Reason: Validation of key "user/sw/lcdproc/lcdd/#0/current/curses/background" with string "purple" failed.
+#> Ingroup: plugin
+#> Module: enum
+#> At: <path>/src/plugins/enum/enum.c:218
+#> Mountpoint: user/sw/lcdproc/lcdd/#0/current
+#> Configfile: <path>/.config/LCDd.conf.17145:1549993505.561040.tmp
+```
+
+### Kdb Editor
+
+Another sophisticated possibility to see available options and edit them
+is to use the `kdb editor`. If you call `kdb editor <key>`, a local editor will
+be opened and all keys under the given `<key>` including all metadata will be shown.
+
+```sh
+kdb editor 'spec/sw/lcdproc/lcdd/#0/current/curses/backlight' ni
```

Comment:
With spec plugin this will not be possible anymore (or restricted). Better to explain how to edit in user/system namespaces.

Reply:
The problem is that you can edit them but do not see the metadata. So the user has to know the allowed values in advance

Reply:
I removed the explanation of the spec namespace

Reply:
> The problem is that you can edit them but do not see the metadata.

Yes, this is a missing feature of `kdb editor`.